envi/archs/arm/disasm.py

```diff
@@ -1726,13 +1785,17 @@ def render(self, mcanv, op, idx):
         mcanv.addNameText(arm_regs[self.shreg][0], typename='registers')
 
     def repr(self, op):
-        rname = arm_regs[self.reg][0]+","
-        return " ".join([rname, shift_names[self.shtype], arm_regs[self.shreg][0]])
+        rname = arm_regs[self.reg][0]+", "
+        if self.display_shift:
+            rname+=shift_names[self.shtype]
+        rname+= arm_regs[self.shreg][0]
+        #return " ".join([rname, shift_names[self.shtype], arm_regs[self.shreg][0]])
+        return rname
 
 class ArmRegShiftImmOper(ArmOperand):
     ''' register shift immediate operand.  see "addressing mode 1 - data processing operands - * shift * by immediate" '''
 
-    def __init__(self, reg, shtype, shimm, va):
+    def __init__(self, reg, shtype, shimm, va, display_shift = True):
```

Comment:
Will look into this further. The purpose of this was the MOV command in the v7 reference (Assuming for UAL reasons) had all variants of mov with a shift become a different operand by the name of the shift. At the time of decoding this the EASIEST method was this toggle. Otherwise have to move it to another operand possibly making a new one that has same exact functionality other than the printing of the shift type. The shift still happens but is not displayed as it becomes redundant.